### ISSUE TITLE
TST: Azure suddenly missing from PR CI lists

### ISSUE THREAD

**Issue by tylerjereddy** [numpy/numpy#12158]
The Azure service is suddenly (and silently !) missing from some recent PRs that don't have much in common and shouldn't do anything that could disable Azure:

#12154
#12157 
#12153

**Comment by charris** [comment]
Yes, I was just looking at that. Either our account was temporary (1 week?) or github notifications have died.

**Comment by charris** [comment]
I was able to manually start a rebuild of an older test, so it looks like notifications have stalled.

**Comment by charris** [comment]
I noticed a stall a couple of days ago, but azure recovered.

**Comment by tylerjereddy** [comment]
cc @chrisrpatterson @ericsciple @jimlamb

**Comment by tylerjereddy** [comment]
Yeah your manual trigger on the set numeric ops deprecate PR was the first run in 7 hours or so.

**Comment by charris** [comment]
Looks like the two coverage tests are also gone, so we are missing three overall.

**Comment by charris** [comment]
Maybe a network problem?

**Comment by chrisrpatterson** [comment]
That is strange.  I see some of the interim PRs have been triggered and report status.  I will send this issue to one of the engineers on the team.

**Comment by tylerjereddy** [comment]
@chrisrpatterson Thanks for checking on this for us

**Comment by lkillgore** [comment]
Hi @tylerjereddy, could you point me to a specific PR / Pipeline pair that you expect should have been triggered?

**Comment by tylerjereddy** [comment]
@lkillgore this is a recent one: #12159 

You can find the pipeline that stopped triggering off PRs 7+ hours ago here: https://dev.azure.com/numpy/numpy/_build?definitionId=4

There's one manual trigger we tried 3h ago, but that's it.

**Comment by charris** [comment]
@lkillgore We don't use the webhook, the recommended method is the app at the github marketplace and is under "Integrations & services" in settings. I checked that and it is still there. What I am wondering is if it was a "try it for one week" sort of thing.

**Comment by charris** [comment]
See https://github.com/marketplace/category/continuous-integration

**Comment by charris** [comment]
Also, being a free plan, there is no payment method. I can see that sort of thing possibly causing problems...

**Comment by lkillgore** [comment]
Hi @charris, the good news is that this is a full product, so it should be working.  I added some logging to your account so I can hopefully get more visibility into what's going on.  Could you please try to open another PR to generate some logs for me to look through?  Thanks!

**Comment by charris** [comment]
Let me try close/reopen on an existing PR, that would previously issue a new request... OK, that did nothing, the old test was still there but it didn't retest. New PR at #12160.

**Comment by tylerjereddy** [comment]
I think @charris will have to do that as an owner of the repo.

**Comment by charris** [comment]
It is configured for all repositories, so none selected. The configuration hasn't changed since this pipeline started running. I can try being selective, but as said, it was working.

**Comment by charris** [comment]
I note that the time it quit working doesn't really fit with someone fooling with the configuration. BTW, we can probably give you permissions if you need them to work on this, although I would have thought you had "root" permissions :)

**Comment by charris** [comment]
Hmm, the last triggered run is interesting, it covered two merges that took place close together. I would have thought the app would queue them up separately by commit hash, but apparently the pipeline just downloads the master branch that is current at run time. Maybe there was some sort of conflict that messed things up?

**Comment by charris** [comment]
I have a feeling that reinstalling the app would fix things, at least temporarily, but the potential problem would still be there, waiting.

**Comment by mattip** [comment]
Now that the Azure team have more extensive logging perhaps getting things going will allow them to find the problem.

**Comment by charris** [comment]
@mattip It's time. @lkillgore I'm going to delete the app from numpy and reinstall it.

**Comment by charris** [comment]
The newly re-installed app isn't triggering either. I think NumPy has been blocked somewhere.

**Comment by charris** [comment]
Note that there doesn't seem to be anyway to use an existing pipeline when the app is installed. Going to turn on appveyor again until this is corrected.

**Comment by charris** [comment]
I think we have reached the "enemy action" stage :) Might be worth checking the nature of the numpy purchase of the app, there might be a built in time limitation, the one week we got still seems suspicious to me.

**Comment by lkillgore** [comment]
Hi @charris, I'm beginning to think your assessment of a time limitation may be correct :( .  Thanks for letting us know about this, and I'll let you know as soon as I have more info on my side.  Thanks!

**Comment by tylerjereddy** [comment]
Azure CI is in the list and running all of a sudden in #12159 after I revised & force-pushed to that branch a few minutes ago.

**Comment by mattip** [comment]
@lkillgore, @chrisrpatterson  the builds seem to be back. Please follow up and let us know what happened so we may close the issue

**Comment by lkillgore** [comment]
Hi @mattip, I patched a bug yesterday; I'm not done testing it yet, but I'll let you know when I'm confident we have this matter resolved.  And, thank you all for your help in drawing our attention to this problem.

**Comment by charris** [comment]
Azure is not currently triggering on the `maintenance/1.15.x` branch, although it is configured to do so.

**Comment by lkillgore** [comment]
Hi @charris, I was just about to write back on this to declare that I think that this bug is fixed.

Something I see that's odd in your definition (and, I really hope that this isn't the problem) is that there's a trailing space in your trigger filter.  Could you try removing that space and see if that fixes it?  If this is what's causing the issue, I'll get that fixed ASAP!

I'll take a look at this further tomorrow.

**Comment by charris** [comment]
@lkillgore The good news is that it wasn't the trailing space. The bad news is that it still doesn't trigger :)

**Comment by charris** [comment]
Looks like merges to the branch are working now.

**Comment by charris** [comment]
Found the problem: The PRs also need an added filter, not just CI. Note that it is not easy to get back to the pipelines when in editing, no easy was to finish with it.

**Comment by charris** [comment]
@tylerjereddy We should add a `trigger:` section to the `*.yml`.

**Comment by charris** [comment]
Going to disable the appveyor builds again, it is just too slow.

**Comment by lkillgore** [comment]
Thanks for letting me know, @charris.  It sounds to me that everything is working, but there could be some improvements in the UX.

**Comment by lkillgore** [comment]
Sure thing!  We added a feature.  Formerly, if your GitHub repo: `numpy/numpy` is using `dev.azure.com/numpy`, then `numpy/numpy2` _has_ to use `dev.azure.com/numpy`.  Now `numpy/numpy2` can be used in a different Azure DevOps account.  Because we stage rollouts of new features to increasing sets of users, not all of our servers had this code yet, so our servers didn't communicate w/ each other correctly which caused the bug.
